charts/opentelemetry-collector/templates/deployment.yaml

```diff
@@ -32,7 +32,11 @@ spec:
   template:
     metadata:
       annotations:
+        {{- if .Values.configMap.existingTemplateFile }}
+        checksum/config: {{ include (print $.Template.BasePath "/" .Values.configMap.existingTemplateFile ) . | sha256sum }}
+        {{- else -}}
         checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
+        {{- end }}
```

Comment:
Can you add a new template in _helpers instead to contain this logic? I expect we'd also need it for daemonset and statefulset.